File: Algorithms/Selection Sort/Selection Sort.py
def selection_sort(arr, n):
    for i in range(n - 1):
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:  # increasing or decreasing order
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]

File: Algorithms/Sorting/Selection Sort/Selection Sort.cpp
template <class T>
void selection_sort(T arr[], int n)
{
    int min_index, temp;
    for (int i = 0; i < n - 1; i++)
    {
        min_index = i;
        for (int j = i + 1; j < n; j++)
        {
            if (arr[j] < arr[min_index]) // increasing or decreasing order
            {
                min_index = j;
            }
        }
        temp = arr[i];
        arr[i] = arr[min_index];
        arr[min_index] = temp;
    }
}